src/fileio.cpp

```diff
@@ -481,6 +482,25 @@ FILE *FioFOpenFile(const char *filename, const char *mode, Subdirectory subdir,
 		strecpy(resolved_name, filename, lastof(resolved_name));
 		strtolower(resolved_name);
 
+		/* Resolve ".." */
+		std::stringstream ss(resolved_name);
+		std::vector<std::string> tokens;
+		for (std::string token; std::getline(ss, token, PATHSEPCHAR);) {
+			if (token == "..") {
+				if (tokens.size() < 4) return nullptr;
+				tokens.pop_back();
+				tokens.pop_back();
+			} else {
+				tokens.push_back(token);
+				tokens.push_back(PATHSEP);
```

Comment:
Maybe instead of storing the PATHSEP in the vector, it could be appended to resolved_name directly?